hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -645,6 +647,59 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	createOrUpdate := r.createOrUpdate(req)
 
+	// Convert old configuration fields to new fields
+
+	if hcluster.Spec.Configuration != nil && len(hcluster.Spec.Configuration.Items) > 0 {
+		originalSpec := hcluster.Spec.DeepCopy()
+		gconfig, err := globalconfig.ParseGlobalConfig(ctx, hcluster.Spec.Configuration)
```

Comment:
nit: It'd nice to have this cohesive code in a unit tested func e.g syncDeprecatedGlobalConfig. That would also keep Reconcile() readable and would prevent it from growing indefinitely.